src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -112,60 +126,53 @@ public String createJwt(
         List<String> roles,
         List<String> backendRoles,
         boolean roleSecurityMode
-    ) throws Exception {
-        final long nowAsMillis = timeProvider.getAsLong();
-        final Instant nowAsInstant = Instant.ofEpochMilli(timeProvider.getAsLong());
-
-        jwtProducer.setSignatureProvider(JwsUtils.getSignatureProvider(signingKey));
-        JwtClaims jwtClaims = new JwtClaims();
-        JwtToken jwt = new JwtToken(jwtClaims);
-
-        jwtClaims.setIssuer(issuer);
-
-        jwtClaims.setIssuedAt(nowAsMillis);
-
-        jwtClaims.setSubject(subject);
-
-        jwtClaims.setAudience(audience);
-
-        jwtClaims.setNotBefore(nowAsMillis);
-
-        if (expirySeconds > maxExpirySeconds) {
-            throw new Exception("The provided expiration time exceeds the maximum allowed duration of " + maxExpirySeconds + " seconds");
+    ) throws JOSEException, ParseException {
+        final Date now = new Date(timeProvider.getAsLong());
+
+        final JWTClaimsSet.Builder claimsBuilder = new JWTClaimsSet.Builder();
+        claimsBuilder.issuer(issuer);
+        claimsBuilder.issueTime(now);
+        claimsBuilder.subject(subject);
+        claimsBuilder.audience(audience);
+        claimsBuilder.notBeforeTime(now);
+
+        if (expirySeconds > MAX_EXPIRY_SECONDS) {
+            throw new IllegalArgumentException(
+                "The provided expiration time exceeds the maximum allowed duration of " + MAX_EXPIRY_SECONDS + " seconds"
+            );
         }
 
-        expirySeconds = (expirySeconds == null) ? defaultExpirySeconds : Math.min(expirySeconds, maxExpirySeconds);
+        expirySeconds = (expirySeconds == null) ? DEFAULT_EXPIRY_SECONDS : Math.min(expirySeconds, MAX_EXPIRY_SECONDS);
         if (expirySeconds <= 0) {
-            throw new Exception("The expiration time should be a positive integer");
+            throw new IllegalArgumentException("The expiration time should be a positive integer");
         }
-        long expiryTime = timeProvider.getAsLong() + expirySeconds;
-        jwtClaims.setExpiryTime(expiryTime);
+        final Date expiryTime = new Date(timeProvider.getAsLong() + expirySeconds * 1000);
```

Comment:
This one right here is not in millisecond: `timeProvider.getAsLong()`. 

You have to change L69:
From `this.timeProvider = () -> System.currentTimeMillis() / 1000;` 
into `this.timeProvider = () -> System.currentTimeMillis();`

Reply:
Hi I'm looking into integration tests